Cargo.toml

```diff
@@ -7,11 +7,25 @@ members = [
 [workspace.package]
 rust-version = "1.74.0"
 
+[workspace.dependencies.soroban-env-host-21]
```

Comment:
Sure, I will do this today. Though one interesting aspect of this on reflection is that since costs are going _down_ in p21->p22, I _think_ it would be comparatively benigh to continue to simulate p22 contracts as p21? It'd just overcharge during the transition.

But I guess more critically if host v21 crashes when exposed to a ledger numbered p22 _and vice-versa_ in this case, then we kinda have to multi-version. We could lie about the protocol number but that would probably be asking for trouble. Generalized support for 2 versions sounds healthier.

Reply:
>  I think it would be comparatively benigh to continue to simulate p22 contracts as p21?

We can't simulate p22 contracts in p21 env, can we? If you meant simulating p21 env/contracts with p22 env, there is a risk of underestimating the cost (as you've pointed out costs are expected to go down in p22).

> Generalized support for 2 versions sounds healthier.

Yes, I do agree that supporting 2 versions of simulation and using them depending on the ledger version (as core does) is a safer and easier to maintain.

Reply:
@dmkozh ok pushed an update that does this correctly (and uses the same shared-module hack core uses).

Ugh, giant change landed on main in the meantime; need to rebase now.